appengine/pond/duck/online/js/online.js

```diff
@@ -46,11 +47,42 @@ Pond.Duck.Online.init = function() {
 
   Pond.Duck.init();
 
+  var duckId = BlocklyGames.getStringParamFromUrl('duckId', null);
+  if (duckId) {
+    Pond.Duck.Datastore.getADuck(duckId, Pond.Duck.Online.updateEditor);
+  }
+
   BlocklyGames.bindClick('duckCreateButton', Pond.Duck.Online.showCreateDuckForm);
   BlocklyGames.bindClick('duckUpdateButton', Pond.Duck.Online.showUpdateDuckForm);
   BlocklyGames.bindClick('duckDeleteButton', Pond.Duck.Online.showDeleteDuckForm);
 };
 
+/**
+ * Callback for when a user loads the page with a duck id.
+ */
+Pond.Duck.Online.updateEditor= function() {
+  var text;
+  if (this.status == 200) {
+    var meta = JSON.parse(this.responseText);
+    var opt_xml = meta['code']['opt_xml'];
+    if (meta['code']['opt_xml']) {
+      Pond.Duck.changeTab(0);
+      Pond.Duck.updateWorkspace(opt_xml);
+    } else {
+      Pond.Duck.changeTab(1);
+      Blockly.utils.dom.addClass(Pond.Duck.editorTabs[0], 'tab-disabled');
+      Blockly.utils.dom.addClass(Pond.Duck.editorTabs[1], 'tab-selected');
+      Blockly.utils.dom.removeClass(Pond.Duck.editorTabs[0], 'tab-selected');
+      Pond.Duck.updateJSCode(meta['code']['js']);
+    }
```

Comment:
use opt_xml for if condition.

I would recommend instead refactoring logic in tabHandler defined in init, the logic for updating the selected tab classes is more robust (this will not work when we add more tabs).

There is a setCode method on BlocklyInterface that works for both xml and js that you could use.

This would function the same, except that we may unecessarily be setting ignoreEditorChanges_
`var opt_xml = meta['code']['opt_xml']`
`var code = opt_xml || meta['code']['js']`
`Pond.Duck.selectTab(opt_xml ? 0 : 1); // would need to be created by refactoring tabHandler`
`Pond.Duck.ignoreEditorChanges_ = false;`
`BlocklyInterface.setCode(code)`
`Pond.Duck.ignoreEditorChanges_ = true;`